pkg/utils/node.go

```diff
@@ -47,19 +47,43 @@ func GetNodeObject(clientset kubernetes.Interface, hostnameOverride string) (*ap
 // Order of preference:
 // 1. NodeInternalIP
 // 2. NodeExternalIP (Only set on cloud providers usually)
-func GetNodeIP(node *apiv1.Node) (net.IP, error) {
+func GetNodeIP(node *apiv1.Node, enableIPv4, enableIPv6 bool) (net.IP, net.IP, error) {
+	var ipAddrv4, ipAddrv6 net.IP
 	addresses := node.Status.Addresses
 	addressMap := make(map[apiv1.NodeAddressType][]apiv1.NodeAddress)
 	for i := range addresses {
 		addressMap[addresses[i].Type] = append(addressMap[addresses[i].Type], addresses[i])
 	}
 	if addresses, ok := addressMap[apiv1.NodeInternalIP]; ok {
-		return net.ParseIP(addresses[0].Address), nil
+		for _, address := range addresses {
+			ipAddr := net.ParseIP(address.Address)
+			if enableIPv4 && ipAddrv4 == nil && len(ipAddr) == net.IPv4len {
```

Comment:
Found an issue here while testing. ipAddr is: ipAddr: net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xa, 0x0, 0xa, 0x7} and thus length is len(ipAddr): 16. Whereas net.IPv4len 4.

I think the best practices is to import "k8s.io/utils/net" and then use the function `IsIPv4(ipAddr)` and `IsIPv6(ipAddr)`
https://pkg.go.dev/k8s.io/utils/net